pkg/platform/types.go

```diff
@@ -181,8 +181,38 @@ type UpdateProjectOptions struct {
 	ProjectConfig ProjectConfig
 }
 
+type DeleteProjectStrategy string
+
+const (
+
+	// DeleteProjectStrategyCascade - delete sub resources along with project leave no orphans behind
+	DeleteProjectStrategyCascade DeleteProjectStrategy = "cascade"
+
+	// DeleteProjectStrategyRestrict - avoid deleting when project contains sub resources (e.g.: functions)
+	DeleteProjectStrategyRestrict DeleteProjectStrategy = "restrict"
+
+	// TODO: add mode to delete sub resources prior to project deletion
+)
+
+func ProjectDeleteStrategyOrRestrict(projectDeleteStrategy string) DeleteProjectStrategy {
+	switch strategy := DeleteProjectStrategy(projectDeleteStrategy); strategy {
+	case DeleteProjectStrategyCascade, DeleteProjectStrategyRestrict:
+		return strategy
+	default:
+
+		// default
+		return DeleteProjectStrategyRestrict
+	}
+}
+
 type DeleteProjectOptions struct {
-	Meta ProjectMeta
+	Meta     ProjectMeta
+	Strategy DeleteProjectStrategy
+
+	// used in conjunction with DeleteProjectStrategyCascade
+	// allowing us to "block" until related resources are (best-effort) removed
+	// allowing testings + clients that requires it (nuctl) to wait for process to complete before terminate
+	WaitForResourcesDeletionCompletion bool
```

Comment:
This is here for testing right?
Please make sure it doesn't leak to production - nuctl/api

Reply:
testing & nuctl.

API dont